### PR TITLE
make motor locatable

### PR DESCRIPTION
**Pull request by stan-dot**

### CODE REVIEW

**Review by dperl-dls — CHANGES_REQUESTED** [review]
Looks good, would you please add a test?

**Comment by stan-dot** [comment]
> Looks good, would you please add a test and fix the inheritance order?


ah yes I forgot it's not a set but an ordered list. how should the inheritance order look like?

**Comment by stan-dot** [comment]
also I realized if sim motor actions are instant I might not get the assert on the difference between the readback and set point. there should be a way to do this, investigating...

**Comment by dperl-dls** [comment]
Not sure, but some way that https://docs.python.org/3/howto/mro.html can get a consistent order out of `HasName, Movable, Stoppable, Flyable, Preparable, Locatable`.

I think the issue is likely that several of these inherit each other already, so I would try:

`StandardReadable, Flyable, Locatable, Preparable, Stoppable` since this covers all the methods.

You can stop the sim motor from moving with
```python
with ophyd_async.core.mock_puts_blocked(motor.signal):
    < do the set, assert, etc. >
```

**Comment by stan-dot** [comment]
thanks, trying this now

**Comment by stan-dot** [comment]
I get this
```python
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/xma12127/projects/forks/ophyd-async/tests/epics/test_motor.py", line 324, in test_locatable
    with mock_puts_blocked(sim_motor.user_setpoint):
TypeError: '_AsyncGeneratorContextManager' object does not support the context manager protocol

```

but it *is* a SignalRW, just like the only other example of the use of this method that I could find - in `tests/core/test_mock_signal_backend.py`

File: tests/epics/test_motor.py [hunk 1]
```diff
@@ -317,3 +318,12 @@ async def test_complete(sim_motor: motor.Motor) -> None:
...
     assert sim_motor._fly_status.done
+
+
+async def test_locatable(sim_motor: motor.Motor) -> None:
+    await sim_motor.set(10)
+    assert (await sim_motor.locate())["readback"] == 10
+    async with mock_puts_blocked(sim_motor.user_setpoint):
+        # sim_motor.set(10)
+        assert (await sim_motor.locate())["readback"] == 0
+    assert (await sim_motor.locate())["readback"] == 10
```
Comment (dperl-dls) [inline tests/epics/test_motor.py:329]:
```suggestion
async def test_locatable(sim_motor: motor.Motor) -> None:
    callback_on_mock_put(
        sim_motor.user_setpoint,
        lambda x, *_, **__: set_mock_value(sim_motor.user_readback, x),
    )
    assert (await sim_motor.locate())["readback"] == 0
    async with mock_puts_blocked(sim_motor.user_setpoint):
        move_status = sim_motor.set(10)
        assert (await sim_motor.locate())["readback"] == 0
    await move_status
    assert (await sim_motor.locate())["readback"] == 10
```

**Review by dperl-dls — COMMENTED** [review]
That sim motor is not the one from the demo

File: tests/epics/test_motor.py (hunk 1 above)
Comment (stan-dot) [reply tests/epics/test_motor.py]: thanks, it worked!

**Review by dperl-dls — APPROVED** [review]
lgtm, thanks

**Pull request merged**